plasma_framework/test/src/utils/BondSize.test.js

```diff
@@ -0,0 +1,43 @@
+const BondSizeMock = artifacts.require('BondSizeMock');
+const { BN } = require('openzeppelin-test-helpers');
+const { expect } = require('chai');
+
+contract('BondSize', () => {
+    beforeEach(async () => {
+        this.initialGasPrice = 20000000000; // 20 gwei
+        this.challengeGasCost = 300000;
+        this.ewmaFactor = 2;
+        this.safetyFactor = 100;
+
+        this.contract = await BondSizeMock.new(
+            this.initialGasPrice,
+            this.challengeGasCost,
+            this.ewmaFactor,
+            this.safetyFactor,
+        );
+
+        this.initialBondSize = new BN(this.initialGasPrice)
+            .muln(this.challengeGasCost)
+            .muln(this.safetyFactor)
+            .divn(100);
+    });
+
+    it('should return the initial bond size', async () => {
+        const bondSize = await this.contract.bondSize();
+        expect(bondSize).to.be.bignumber.equal(this.initialBondSize);
+    });
+
+    it('should lower the bond size when average gas price decreases', async () => {
```

Comment:
Would we not like to test with the `ewma` formula or we'd only care about increase/decrease?

Reply:
Hmm, I think it's better that the tests don't know the exact formula, it kind of makes them brittle

Reply:
Okay, so what we want to get from this bond library is more of an "average" concept, and we don't really care about what average algo/formula we are using. Just want to make sure I have your intention right.